doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    ROS_INFO_NAMED("tutorial", "Adding an object into the world");
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Attach Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    planning_scene_interface.addCollisionObjects(collision_objects);
+  }
+
+  void attach_parcel()
+  {
+    visual_tools.prompt("Press 'next' to attach the object");
+
+    // BEGIN_SUB_TUTORIAL attach_obj
+    // Attaching and detaching objects
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // Attaching or detaching an already existing object is easy.
+    //
+    // To attach an object.
+    move_group.attachObject(collision_objects.back().id);
+    // END_SUB_TUTORIAL
+  }
+
+  void plan_parcel()
+  {
+    // Rviz
+    visual_tools.prompt("Press 'next' to plan and move with the attached object");
+
+    // target_pose repositioning and planning
+    target_pose.position.y += 0.2;
+    target_pose.position.z += 0.2;
+    move_group.setPoseTarget(target_pose);
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 7 (pose goal with attached cuboid) %s", success ? "" : "FAILED");
+
+    // Rviz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishAxisLabeled(target_pose, "point");
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.publishText(text_pose, "Pose Goal with Attached Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+  }
+
+  void detach_parcel()
+  {
+    ROS_INFO_NAMED("tutorial", "Detach the object from the robot");
+
+    // BEGIN_SUB_TUTORIAL detach_obj
+    // To detach an object.
+    move_group.detachObject(collision_objects.back().id);
+    // END_SUB_TUTORIAL
+
+    // RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Object Detached", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+  }
+
+  void remove_parcel()
+  {
+    visual_tools.prompt("Press 'next' to remove the object");
+
+    // Remove the collision object from the world.
+    ROS_INFO_NAMED("tutorial", "Remove the object from the world");
+    planning_scene_interface.removeCollisionObjects(object_ids);
+    object_ids.pop_back();
+    collision_objects.pop_back();
+
+    // Rviz
+    visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to complete the demo");
+  }
+
+  void move()
+  {
+    // BEGIN_SUB_TUTORIAL movement
+    // Moving the robot
+    // ^^^^^^^^^^^^^^^^
+    //
+    // We can now use the plan we just created with movegroup to move the robot.
+    move_group.execute(my_plan);
+    /* move_group.move(); // calls plan and execute at the same time */
+
+    // Keeping the startstate updated with the robot is important. If the startstate isn't where
+    // the robot is then it will be unable to move.
+    move_group.setStartState(*move_group.getCurrentState());
```

Comment:
Check the API. I believe you only need to do this to plan from a start state other than the current state. Otherwise you should just clear the start state and it will plan from the current state by default.

Also, this shouldn't be part of the execute command. execute should only execute. secondary effects like this are bad practice. If you want to set the start state, do it before planning. not after execution